knative-operator/pkg/common/service_monitor.go

```diff
@@ -54,23 +81,210 @@ func SetupServiceMonitor(ctx context.Context, cfg *rest.Config, metricsPort int3
+			Version: "v1alpha1",
+			Kind:    "KnativeEventing",
+		},
+	}
+	// To generate metrics in other namespaces, add the values below.
+	// This is due to this bug: https://github.com/operator-framework/operator-sdk/issues/2494
+	// For the workaround check here: https://github.com/operator-framework/operator-sdk/pull/2601/files#r396745465
+	// and https://github.com/shipwright-io/build/pull/73
+	// In order to avoid getting a bad value we avoid using k8sutil.GetWatchNamespace() that gets the value from the WATCH_NAMESPACE
+	// env var. That value is by default "" but user may change it, affecting the metrics endpoint.
+	namespaces := []string{""}
+	// Generate and serve custom resource specific metrics.
+	err := kubemetrics.GenerateAndServeCRMetrics(cfg, namespaces, gvkFilterList, metricsHost, operatorMetricsPort)
 	if err != nil {
 		return err
 	}
-	// Get the namespace the operator is currently deployed in.
-	operatorNs, err := k8sutil.GetOperatorNamespace()
+	return nil
+}
+
+func SetupEventingBrokerServiceMonitors(client client.Client, namespace string, instance *eventingv1alpha1.KnativeEventing) error {
+	if err := isServiceMonitorAvailable(); err != nil {
+		return err
+	}
+	manifest, err := mf.NewManifest(getMonitorPath(TestEventingBrokerServiceMonitorPath, EventingBrokerServiceMonitorPath), mf.UseClient(mfclient.NewClient(client)))
 	if err != nil {
+		return fmt.Errorf("unable to parse broker service monitors: %w", err)
+	}
+	transforms := []mf.Transformer{mf.InjectOwner(instance), mf.InjectNamespace(namespace)}
+	if manifest, err = manifest.Transform(transforms...); err != nil {
+		return fmt.Errorf("unable to transform broker service monitors manifest: %w", err)
+	}
+	// this is required because Apply will fail with not known resource later on
+	monitor := &monitoringv1.ServiceMonitor{}
+	var SchemeGroupVersion = schema.GroupVersion{Group: "monitoring.coreos.com", Version: "v1"}
+	scheme.Scheme.AddKnownTypes(SchemeGroupVersion, monitor)
+	if err := manifest.Apply(); err != nil {
 		return err
 	}
-	// To generate metrics in other namespaces, add the values below.
-	ns := []string{operatorNs}
-	// Generate and serve custom resource specific metrics.
-	err = kubemetrics.GenerateAndServeCRMetrics(cfg, ns, filteredGVK, metricsHost, operatorMetricsPort)
+	return nil
+}
+
+func SetupSourceServiceMonitor(client client.Client, namespace string, instance *appsv1.Deployment) error {
+	var sLabel, sNameLabel, sRoleLabel string
+
+	if err := isServiceMonitorAvailable(); err != nil {
+		return err
+	}
+
+	monitor := &monitoringv1.ServiceMonitor{}
+	var SchemeGroupVersion = schema.GroupVersion{Group: "monitoring.coreos.com", Version: "v1"}
+	scheme.Scheme.AddKnownTypes(SchemeGroupVersion, monitor)
+	labels := instance.Spec.Selector.MatchLabels
+
+	if l, ok := labels[SourceLabel]; ok {
+		sLabel = l
+	}
+
+	if l, ok := labels[SourceNameLabel]; ok {
+		sNameLabel = l
+	}
+
+	if l, ok := labels[SourceRoleLabel]; ok {
+		sRoleLabel = l
+	}
```

Comment:
The if's are redundant. Might as well just write:

```golang
var (
  sLabel = labels[SourceLabel]
  sNameLabel = labels[SourceNameLabel]
  sRoleLabel = abels[SourceRoleLabel]
)
```

For the same outcome.

Reply:
ok will update.